design/host-cgroups.md

```diff
@@ -200,9 +204,132 @@ utilized unaccounted for on the host.
+│   └── tasks
+├── cpuset
+│   ├── cgroup.procs
+│   └── tasks
+├── devices
+│   ├── cgroup.procs
+│   └── tasks
+├── freezer
+│   ├── cgroup.procs
+│   └── tasks
+├── hugetlb
+│   ├── cgroup.procs
+│   └── tasks
+├── memory
+│   ├── cgroup.procs
+│   └── tasks
+├── net_cls -> net_cls,net_prio
+├── net_cls,net_prio
+│   ├── cgroup.procs
+│   └── tasks
+├── net_prio -> net_cls,net_prio
+├── perf_event
+│   ├── cgroup.procs
+│   └── tasks
+├── pids
+│   ├── cgroup.procs
+│   └── tasks
+└── systemd
+    ├── cgroup.procs
+    └── tasks
+```
+
+A whole process can join a cgroup by writing its process id (`pid`) to `cgroup.procs` file,
+or join a cgroup partially by writing the task (thread) id (`tid`) to `tasks` file.
+
+Kata containers support `v1` by default and no change in the configuration file is needed.
+To know more about `cgroups v1`, see [cgroupsv1(7)][2]
+
+## Cgroups V2
+
+`Cgroups v2` are also known as unified cgroups, Unlike `cgroups v1`, the cgroups are
+mounted under the same cgroup filesystem. `Cgroups v2` hierarchy may look like the following
+diagram:
+
+```
+/sys/fs/cgroup/system.slice
+├── cgroup.controllers
+├── cgroup.events
+├── cgroup.freeze
+├── cgroup.max.depth
+├── cgroup.max.descendants
+├── cgroup.procs
+├── cgroup.stat
+├── cgroup.subtree_control
+├── cgroup.threads
+├── cgroup.type
+├── cpu.max
+├── cpu.pressure
+├── cpu.stat
+├── cpu.weight
+├── cpu.weight.nice
+├── io.bfq.weight
+├── io.latency
+├── io.max
+├── io.pressure
+├── io.stat
+├── memory.current
+├── memory.events
+├── memory.events.local
+├── memory.high
+├── memory.low
+├── memory.max
+├── memory.min
+├── memory.oom.group
+├── memory.pressure
+├── memory.stat
+├── memory.swap.current
+├── memory.swap.events
+├── memory.swap.max
+├── pids.current
+├── pids.events
+└── pids.max
+```
+
+Same as `cgroups v1`, a whole process can join the cgroup by writing its process id (`pid`) to
+`cgroup.procs` file, or join a cgroup partially by writing the task (thread) id (`tid`) to `cgroup.threads` file.
+
+Kata containers don't support `v2` by default, `sandbox_cgroup_only` must be set to `true` in
+the configuration file.
+To know more about `cgroups v2`, see [cgroupsv2(7)][3]
+
+### Limitations
+
+Officially `cgroups v2` are not yet supported in the guest, but they can work with a fedora 31 rootfs
+image and the latest kata agent (at the time this document was written the latest version
+was `1.11.0-alpha0`). For more information about the status of this feature see [issue #2494][4]
+
+# Summary
+
+| cgroup option | default? | status | pros | cons | cgroups
+|-|-|-|-|-|-|
+| `SandboxCgroupOnly=false` | yes | legacy | Easiest to make Kata work | Unaccounted for memory and resource utilization | v1
+| `SandboxCgroupOnly=true` | no | recommended | Complete tracking of Kata memory and CPU utilization. In Kubernetes, the Kubelet can fully constrain Kata via the pod cgroup | Requires upper layer orchestrator which sizes sandbox cgroup appropriately | v1, v2
```

Comment:
I'm struggling with the status of `recommended`, when it is not the default today. Maybe we can change that to `supported`, and if we have a timeline for when we might make it the default, add that in as well?

Reply:
I agree, we are waiting for the k8s podoverhead feature that may land in the next alpha release.... ?

cc @amshinde @egernst

Reply:
Correct.  The K8S PodOverhead feature [should move to beta in 1.18](https://github.com/kubernetes/kubernetes/issues/88339).  I wouldn't want to move to a default until this is in place, and until we have appropriate RuntimeClass definitions available which utilize Overhead.